Clarify MariaDB (#5451)
mattermost/docs: source/install/software-hardware-requirements.rst

Amazon Aurora equivalents of both MySQL and PostgreSQL are also supported.

> Maria v10+ no longer functions as a MySQL drop-in replacement which means it is not supported for Mattermost due to the requirement of MySQL 5.7.12. Prior versions were not officially supported but may have functioned in older Mattermost releases. If you are running MariaDB now, migrating from Maria to the MySQL equivalent is suggested.

Reviewer: ```suggestion
MariaDB v10+ no longer functions as a MySQL drop-in replacement, and it's not supported for Mattermost due to the requirement of MySQL 5.7.12. Prior versions of MariaDB were not officially supported but may have functioned in older Mattermost releases. If you are running MariaDB now, migrating from MariaDB to the MySQL equivalent is suggested.
```

Reviewer: Suggested or recommended?

Author: I mean, it's the **only** real option they have. I was just trying to be nice.

Reviewer: Updated the verbiage to recommended. :)
